pkg/storage/storage.go

```diff
@@ -17,6 +17,14 @@ type ReadSeekCloser interface {
 	io.Closer
 }
 
+// Uploader upload file with chunks.
+type Uploader interface {
+	// UploadPart upload part of file data to storage
+	UploadPart(ctx context.Context, data []byte, partNum int) error
```

Comment:
what is the purpose of `partNum`? 

as `S3Uploader` does not support running `UploadPart` concurrently, the number can be set as just `len(s.completeParts)+1`.